File: utils/src/main/java/com/natchuz/hub/utils/mojang/MojangAPI.java
package com.natchuz.hub.utils.mojang;

import java.util.UUID;

public interface MojangAPI {

    /**
     * Returns username of user with specified UUID
     */
    String getUsername(UUID uuid);

    /**
     * Returns UUID of user with specified username
     */
    UUID getUUID(String username);

}

File: utils/src/main/kotlin/com/natchuz/hub/utils/mojang/MojangAPI.kt
package com.natchuz.hub.utils.mojang

import java.util.*

interface MojangAPI {
    /**
     * Returns username of user with specified UUID
     */
    fun getUsername(uuid: UUID): String?

    /**
     * Returns UUID of user with specified username
     */
    fun getUUID(username: String): UUID?
}